be/src/storage/lake/update_manager.cpp

```diff
@@ -497,15 +499,19 @@ size_t UpdateManager::get_rowset_num_deletes(int64_t tablet_id, int64_t version,
     return num_dels;
 }
 
-Status UpdateManager::publish_primary_compaction(const TxnLogPB_OpCompaction& op_compaction,
+Status UpdateManager::publish_primary_compaction(const TxnLogPB_OpCompaction& op_compaction, int64_t txn_id,
                                                  const TabletMetadata& metadata, Tablet tablet, IndexEntry* index_entry,
                                                  MetaFileBuilder* builder, int64_t base_version) {
     auto& index = index_entry->value();
     // 1. iterate output rowset, update primary index and generate delvec
     std::shared_ptr<TabletSchema> tablet_schema = std::make_shared<TabletSchema>(metadata.schema());
     RowsetPtr output_rowset =
             std::make_shared<Rowset>(tablet, std::make_shared<RowsetMetadata>(op_compaction.output_rowset()));
-    auto compaction_state = std::make_unique<CompactionState>(output_rowset.get(), this);
+    auto compaction_entry = _compaction_cache.get_or_create(strings::Substitute("$0_$1", tablet.id(), txn_id));
```

Comment:
Wrap the construction of the cache key in a separate function

Reply:
Done